tests/integration/test_metadata_model.py

```diff
@@ -54,31 +143,70 @@ def test_submit_filebased_manifest(
-        manifest_path = helpers.get_data_path(manifest_path)
+        # AND a filebased test manifest
+        load_args = {"dtype": "string"}
+        manifest = helpers.get_data_frame(
+            manifest_path, preserve_raw_input=True, **load_args
+        )
+        manifest_full_path = helpers.get_data_path(manifest_path)
 
-        # WHEN the manifest it submitted
+        # WHEN the manifest is submitted and files are annotated
         # THEN submission should complete without error
         with does_not_raise():
             manifest_id = meta_data_model.submit_metadata_manifest(
-                manifest_path=manifest_path,
+                manifest_path=manifest_full_path,
                 dataset_id=dataset_id,
-                manifest_record_type="file_and_entities",
+                manifest_record_type=manifest_record_type,
                 restrict_rules=False,
                 file_annotations_upload=True,
                 hide_blanks=False,
                 validate_component=validate_component,
                 dataset_scope=dataset_scope,
             )
 
-            # AND the manifest should be submitted to the correct place
-            assert manifest_id == expected_manifest_id
+        # AND the files should be annotated
+        spy_add_annotations.assert_called_once()
 
-            # AND the manifest should be uploaded as a CSV
-            spy_upload_file_as_csv.assert_called_once()
-            # AND annotations should be added to the files
-            spy_add_annotations.assert_called_once()
+        # AND the annotations on the entities should have the correct metadata
+        for index, row in manifest.iterrows():
+            entityId = row["entityId"]
+            expected_sample_id = row["Sample ID"]
+            annos = synapse_store.syn.get_annotations(entityId)
+            sample_id = annos["SampleID"][0]
+            assert str(sample_id) == str(expected_sample_id)
+
+        # AND the annotations on the manifest file itself should have the correct metadata
+        manifest_annos = synapse_store.syn.get_annotations(manifest_id)
+        for annotation in manifest_annos.keys():
+            if annotation == "eTag":
+                continue
+            if annotation == "entityId":
+                assert manifest_annos[annotation][0] == expected_manifest_id
+            else:
+                assert manifest_annos[annotation][0] == manifest[annotation].unique()
+
+        # AND the annotations in the manifest table should reflect the ones in the file, if applicable
+        if manifest_record_type == "table_and_file":
+            table_annotations = synapse_store.syn.tableQuery(
+                f"select * from {expected_table_id}", downloadLocation=download_dir
+            ).asDataFrame()
+
+            table_columns = table_annotations.columns
+            manifest_columns = [col.replace(" ", "") for col in manifest.columns]
+            assert set(table_columns) == set(manifest_columns)
```

Comment:
Thanks for the update Jenny! I think here you make sure that the table being submitted has the same columns as the columns in the manifest. But my previous comment also mentioned about the annotations of the table: 

<img width="994" alt="Screenshot 2024-09-17 at 4 09 21 PM" src="https://github.com/user-attachments/assets/cdcfa589-83d2-429e-844e-99859c19f430">

Do you have time to add a test here and make sure the following?  
* the table gets annotated with the correct entity id 
* etag is not empty

If not, no worry. We could open an additional ticket 👍🏼

Reply:
[Added](https://github.com/Sage-Bionetworks/schematic/pull/1485/commits/f32e756e40efdb440ccefb6c8f5bf8f9b84c6e1c)!